src/main/java/ro/hiringsystem/service/impl/AnonymousUserServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package ro.hiringsystem.service.impl;
+
+import ro.hiringsystem.model.AnonymousUser;
+import ro.hiringsystem.service.AnonymousUserService;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class AnonymousUserServiceImpl implements AnonymousUserService {
```

Comment:
After the modifications proposed for the interface of the service, you can either implement UserService<AnonymousUser> or if you extend it in a custom interface, implement that one.